README.md

```diff
@@ -1,3 +1,517 @@
+       @Override
+       protected List<ReactPackage> getPackages() {
+           return Arrays.<ReactPackage>asList(
+               new MainReactPackage(),
+               new RNDateTimePickerPackage() // <-- add this line
+           );
+       }
+   }
+   ```
+
+## Tests
+
+### Jest
+
+```
+$ npm install
+$ npm run test
+```
+
+### Detox
+
+Detox is a gray box end-to-end testing and automation library for mobile apps.
+- [Dependencies required](https://github.com/wix/Detox/blob/master/docs/Introduction.GettingStarted.md#step-1-install-dependencies)
+
+For cleaning all the detox builds just run `npm run detox:clean`.
+
+#### iOS
+
+- debug:
+  ```sh
+  npm install
+  # Debug requires to run Metro Bundler
+  npm run start
+  npm run detox:ios:build:debug
+  npm run detox:ios:test:debug
+  ```
+
+- release:
+  
+  ```sh
+  npm install
+  npm run detox:ios:build:release
+  npm run detox:ios:test:release
+  ```
+
+#### Android
+
+  An existing Android emulator is required to match the name defined in `detox.configurations.android.emu.debug.name` and `detox.configurations.android.emu.release.name` inside the `package.json`.
+
+  - debug:
+    ```sh
+    npm install
+    # Debug requires to run Metro Bundler
+    npm run start
+    npm run detox:android:build:debug
+    npm run detox:android:test:debug
+    ```
+
+  - release:
+    
+    ```sh
+    npm install
+    npm run detox:android:build:release
+    npm run detox:android:test:release
+    ```
+
+## Running the example app
+
+1. Run `npm install`
+2. Run `npm run start` for starting Metro Bundler
+3. Run `npm run start:ios` or `npm run start:android`
+
+## General Usage
+
+### Importing it
+
+```js
+import DateTimePicker from 'react-native-datetimepicker';
+```
+or
+
+```js
+var DateTimePicker = require('react-native-datetimepicker');
+```
+
+### Basic usage
+
+```js
+<RNDateTimePicker 
+  value={date} 
+  mode={mode} 
+  is24Hour={true}
+  display="default" 
+  onChange={this.setDate}
+/>
+```
+
+### Props
+
+#### `mode` (`optional`)
+
+Defines the type of the picker.
+
+List of possible values:
+- `"date"` (default for `iOS` and `Android)
+- `"time"`
+- `"datetime"` (`iOS` only)
+- `"countdown"` (`iOS` only)
+
+```js
+<RNDateTimePicker mode="time" />
+```
+
+#### `display` (`optional`, `Android only`)
+
+Defines the visual display of the picker for Android and will be ignored for iOS.
+
+List of possible values:
+- `"default"`
+- `"spinner"`
+- `"calendar"` (only for `date` mode)
+- `"clock"` (only for `time` mode)
+
+```js
+<RNDateTimePicker display="spinner" } />
+```
+
+#### `onChange` (`optional`)
+
+Date change handler.
+
+This is called when the user changes the date or time in the UI. It receives the event and the date as parameters.
+
+```js
+setDate = (event, date) => {}
+
+<RNDateTimePicker onChange={this.setDate} />
+```
+
+#### `value` (`required`)
+
+Defines the date or time value used in the component.
+
+```js
+<RNDateTimePicker value={new Date()} />
```

Comment:
This is the default for `value`, specifying a different value or an example with state/ HOC props would be nice to have.